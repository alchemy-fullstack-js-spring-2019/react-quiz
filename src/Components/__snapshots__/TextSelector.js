import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class TextSelector extends 