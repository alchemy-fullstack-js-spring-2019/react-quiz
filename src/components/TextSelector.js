import React, { PureComponent } from 'react';

export default class TextSelector extends PureComponent {

  state = {
    text: ''
  }

  changeHandler = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { text } = this.state;

    return (
      <>
      <input name="text" type="text" value={ text } onChange={this.changeHandler} />
      </>
    );
  }
}
