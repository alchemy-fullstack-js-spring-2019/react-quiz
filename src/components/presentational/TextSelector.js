import React from 'react';
import PropTypes from 'prop-types';

export default function TextSelector({ text, handleChange }) {
  return <input name="text" value={text} onChange={handleChange} />;
}

TextSelector.propTypes = {
  text: PropTypes.string,
  handleChange: PropTypes.func
};
