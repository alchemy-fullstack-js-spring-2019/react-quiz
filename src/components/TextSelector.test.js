import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('Color input component', () => {
  it('renders a ColorSelector', () => {
    const wrapper = shallow(<TextSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
