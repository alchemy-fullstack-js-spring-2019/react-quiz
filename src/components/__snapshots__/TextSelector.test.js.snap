// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color input component renders a ColorSelector 1`] = `
<Fragment>
  <input
    name="text"
    type="text"
    value=""
  />
</Fragment>
`;
